File: generated-code/haskell/codecraft/FileReadWrite/src/Vec2Float.hs
module Vec2Float where

import Prelude hiding (id)
import qualified Trans
import Trans (Trans)


-- | 2 dimensional vector.
data Vec2Float = Vec2Float {
    -- | `x` coordinate of the vector
    x :: Float,
    -- | `y` coordinate of the vector
    y :: Float }
    deriving Show

instance Trans Vec2Float where
    read = do
        x <- Trans.read
        y <- Trans.read
        return Vec2Float {
            x,
            y }

    write Vec2Float {
        x,
        y } = do
            Trans.write x
            Trans.write y

File: generated-code/php/codecraft/FileReadWrite/Vec2Float.php
<?php

namespace  {
    require_once 'Stream.php';

    /**
     * 2 dimensional vector.
     */
    class Vec2Float
    {
        /**
         * `x` coordinate of the vector
         */
        public float $x;
        /**
         * `y` coordinate of the vector
         */
        public float $y;
    
        function __construct(float $x, float $y)
        {
            $this->x = $x;
            $this->y = $y;
        }
    
        /**
         * Read Vec2Float from input stream
         */
        public static function readFrom(\InputStream $stream): Vec2Float
        {
            $x = $stream->readFloat32();
            $y = $stream->readFloat32();
            return new Vec2Float($x, $y);
        }
        
        /**
         * Write Vec2Float to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeFloat32($this->x);
            $stream->writeFloat32($this->y);
        }
    }
}